snap/types.go

```diff
@@ -140,6 +140,48 @@ func (confinementType *ConfinementType) fromString(str string) error {
 	return nil
 }
 
+// GradeType represents the grade of the snap
+type GradeType string
+
+const (
+	DevelGrade  GradeType = "devel"
+	StableGrade GradeType = "stable"
+	EmptyGrade  GradeType = ""
+)
+
+// UnmarshalJSON sets *gradeType to a copy of data, assuming validation passes
+func (gt *GradeType) UnmarshalJSON(data []byte) error {
```

Comment:
Please drop those and implement a validating TextMarshaler, those will be used in json automatically but have the advantage of being simpler and more generic.